packages/fast-components-react-base/src/listbox/listbox.tsx

```diff
@@ -0,0 +1,600 @@
+        return (
+            element instanceof HTMLElement &&
+            element.getAttribute("role") === "option" &&
+            !this.isDisabledElement(element)
+        );
+    };
+
+    /**
+     * Determines if a given element is disabled
+     */
+    private isDisabledElement = (element: Element): element is HTMLElement => {
+        return (
+            element instanceof HTMLElement &&
+            element.getAttribute("aria-disabled") === "true"
+        );
+    };
+
+    /**
+     * Return an array of all focusabled elements that are children
+     * of the context menu
+     */
+    private domChildren(): Element[] {
+        return canUseDOM() && this.rootElement.current instanceof HTMLElement
+            ? Array.from(this.rootElement.current.children)
+            : [];
+    }
+
+    /**
+     * Sets focus to the nearest focusable element to the supplied focusIndex.
+     * The adjustment controls how the function searches for other focusable elements
+     * if the element at the focusIndex is not focusable. A positive number will search
+     * towards the end of the children array, whereas a negative number will search towards
+     * the beginning of the children array.  Returns the focussed item id or an empty string
+     * if none found
+     */
+    private setFocus(focusIndex: number, adjustment: number): string {
+        const children: Element[] = this.domChildren();
+        let focusItemId: string = "";
+
+        while (inRange(focusIndex, children.length)) {
+            const child: Element = children[focusIndex];
+            focusItemId = child.id;
+            if (this.isFocusableElement(child)) {
+                child.focus();
+                break;
+            }
+
+            focusIndex += adjustment;
+        }
+
+        return focusItemId;
+    }
+
+    /**
+     * Function called by child select options when they have been focused
+     */
+    private listboxItemfocused = (
+        item: ListboxItemData,
+        event: React.FocusEvent<HTMLDivElement>
+    ): void => {
+        const target: Element = event.currentTarget;
+        const focusIndex: number = this.domChildren().indexOf(target);
+
+        if (this.isDisabledElement(target)) {
+            target.blur();
+            return;
+        }
+
+        this.setState({
+            focusIndex,
+            focussedItemId: item.id,
+        });
+
+        if (!this.props.multiselectable) {
+            this.updateSelection([item]);
+        }
+    };
+
+    /**
+     * Handle the keydown event of the root menu
+     */
+    private handleMenuKeyDown = (event: React.KeyboardEvent<HTMLDivElement>): void => {
+        if (event.defaultPrevented) {
+            return;
+        }
+
+        let focusItemId: string;
+
+        switch (event.keyCode) {
+            case KeyCodes.arrowDown:
+            case KeyCodes.arrowRight:
+                event.preventDefault();
```

Comment:
nit: you keep calling event.preventDefault() in every switch case, you can just put this before the switch.